### PR TITLE
fix: detect casing in modifier keys

### LINKED ISSUE #75: Casing Matters in Dupes
if you have `<C-a>` and `<c-a>` mapped. This isn't detected as a dupe

### PR DESCRIPTION
**Pull request by tris203**
closes #75

### CODE REVIEW

File: lua/hawtkeys/utils.lua [hunk 1]
```diff
@@ -127,4 +128,14 @@ function M.reduceHome(path)
     return reduced
 end
 
+---@param lhs string
+---@return string
+function M.sanitise_modifier_keys(lhs)
+    --- to lower anything in < >
+    local santisedLhs = lhs:gsub("<[^>]*>", function(match)
+        return match:lower()
+    end)
```
Comment (willothy) [inline lua/hawtkeys/utils.lua]:
```suggestion
    local santisedLhs = lhs:gsub("<[^>]*>", string.lower)
```

**Review by willothy — CHANGES_REQUESTED** [review]
Small change then lgtm - capital letters in pseudo keys need to be mapped with shift right? (`<C-S-d>` instead of `<C-D>`)

File: lua/hawtkeys/utils.lua (hunk 1 above)
Comment (willothy) [reply lua/hawtkeys/utils.lua]: hmm or maybe we should just sanitize the mappings before storing them instead of in during the search?

**Comment by tris203** [comment]
> Small change then lgtm - capital letters in pseudo keys need to be mapped with shift right? (`<C-S-d>` instead of `<C-D>`)

yeah thats what we found.

We currently have the same issue in search, so i need to do some work there.

Maybe storing them with the altered casing is the move, but i like the idea of presenting them however the user has them recorded not how we handle them internally?

**Review by willothy — APPROVED** [review]
lgtm, though we probably want to optimize this a bit later on or store the raw keymaps.

**Pull request merged**